### PR TITLE
Add a wipeLocal function for logins, fixes #493

### LINKED ISSUE #493: Add a wipeLocal method to logins storage
Should delete all local data, leaving remote unchanged (returning to the state before first sync), requested by lockbox.

### PR DESCRIPTION
**Pull request by thomcc**
This applies on top of #495

### CODE REVIEW

**Comment by thomcc** [comment]
A-C side of this is https://github.com/thomcc/android-components/tree/logins-wipelocal-and-locking

File: components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt [hunk 1]
```diff
@@ -12,50 +12,119 @@ class SyncUnlockInfo (
 )
 
 interface LoginsStorage : AutoCloseable {
-
+    /**
+     * Lock (close) the database.
+     *
+     * @throws [MismatchedLockException] if the database is already locked
```
Comment (vladikoff) [inline components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt]: should there be a LoginsStorageException here as well?

File: components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt [hunk 2]
```diff
@@ -12,50 +12,119 @@ class SyncUnlockInfo (
...
      * Fetch the full list of passwords from the underlying storage layer.
+     *
+     * @throws [LoginsStorageException] On unexpected errors (IO failure, rust panics, etc)
      */
+    @Throws(LoginsStorageException::class)
     fun list(): List<ServerPassword>
 
     /**
-     * Insert the provided login into the database.
+     * Inserts the provided login into the database, returning it's id.
```
Comment (vladikoff) [inline components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt]: `its id`?

File: components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt (hunk 1 above)
Comment (thomcc) [reply components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt]: Nope!

Comment (vladikoff) [reply components/logins/android/src/main/java/mozilla/appservices/logins/LoginsStorage.kt]: Ok 👍

**Pull request merged**